src/components/Authorize/Authorize.js

```diff
@@ -11,26 +11,54 @@ const Authorize = () => {
       .substring(1)
       .split('access_token=')[1]
       .split('&')[0];
-    var options = {
-      url: 'https://api.spotify.com/v1/me',
-      headers: { Authorization: 'Bearer ' + access_token },
-      json: true,
-    };
+    console.log(access_token);
+    let state = window.location.hash
+      .substring(1)
+      .split('state=')[1];
+    if(state === localStorage.getItem("stateValue")){
+     var options = {
+        url: 'https://api.spotify.com/v1/me',
+        headers: { Authorization: 'Bearer ' + access_token },
+        json: true,
+      };
 
-    // use the access token to access the Spotify Web API
-    request.get(options, function (error, response, body) {
-      console.log(body);
+      // use the access token to access the Spotify Web API
+      request.get(options, function (error, response, body) {
+        if(error) {
+          localStorage.removeItem("stateValue")
+          return setUserData({
+            ...userData,
+            isLoggedIn: "error",
+            data: {}
+          });
```

Comment:
This doesn't work at all, I tried login with your client id and my account that doesn't have access to client id. I get routed to `/home` even though I get this in the console:
<img src='https://puu.sh/IutQQ/149ebf8e9e.png' width='800' />